libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,521 @@
+    valid_response_codes = [httplib.OK, httplib.ACCEPTED, httplib.CREATED,
+                            httplib.NO_CONTENT]
+
+    def parse_body(self):
+        if not self.body:
+            return None
+        # json.loads doesn't like the regex expressions used in godaddy schema
+        self.body = self.body.replace('\\.', '\\\\.')
+
+        data = json.loads(self.body)
+        return data
+
+    def parse_error(self):
+        data = self.parse_body()
+        raise GoDaddyDNSException(
+            data['code'], data['message'])
+
+    def success(self):
+        return self.status in self.valid_response_codes
+
+
+class GoDaddyDNSConnection(ConnectionKey):
+    responseCls = GoDaddyDNSResponse
+    host = API_ROOT
+
+    allow_insecure = False
+
+    def __init__(self, key, secret, shopper_id, secure=True, host=None,
+                 port=None, url=None, timeout=None,
+                 proxy_url=None, backoff=None, retry_delay=None):
+        super(GoDaddyDNSConnection, self).__init__(
+            key,
+            secure=secure, host=host,
+            port=port, url=url,
+            timeout=timeout,
+            proxy_url=proxy_url,
+            backoff=backoff,
+            retry_delay=retry_delay)
+        self.key = key
+        self.secret = secret
+        self.shopper_id = shopper_id
+
+    def add_default_headers(self, headers):
+        headers['X-Shopper-Id'] = self.shopper_id
+        headers['Authorization'] = "sso-key %s:%s" % \
+            (self.key, self.secret)
+        return headers
+
+
+class GoDaddyDNSDriver(DNSDriver):
+    """
+    A driver for GoDaddy DNS.
+
+    This is for customers of GoDaddy
+    who wish to purchase, update existing domains
+    and manage records for DNS zones owned by GoDaddy NS servers.
+    """
+    type = Provider.GODADDY
+    name = 'GoDaddy DNS'
+    website = 'https://www.godaddy.com/'
+    connectionCls = GoDaddyDNSConnection
+
+    RECORD_TYPE_MAP = {
+        RecordType.A: 'A',
+        RecordType.AAAA: 'AAAA',
+        RecordType.CNAME: 'CNAME',
+        RecordType.MX: 'MX',
+        RecordType.NS: 'SPF',
+        RecordType.SRV: 'SRV',
+        RecordType.TXT: 'TXT',
+    }
+
+    def __init__(self, shopper_id, key, secret,
+                 secure=True, host=None, port=None):
+        """
+        Instantiate a new `GoDaddyDNSDriver`
+
+        :param  shopper_id: Your customer ID or shopped ID with GoDaddy
+        :type   shopper_id: `str`
+
+        :param  key: Your access key from developer.godaddy.com
+        :type   key: `str`
+
+        :param  secret: Your access key secret
+        :type   secret: `str`
+        """
+        super(GoDaddyDNSDriver, self).__init__(key=key, secret=secret,
+                                               secure=secure,
+                                               host=host, port=port,
+                                               shopper_id=str(shopper_id))
+
+    def _to_zones(self, items):
```

Comment:
Minor code style thing - please put "'private" methods to the bottom - https://libcloud.readthedocs.org/en/latest/development.html#function-and-method-ordering